app/recipe/tests/test_recipe_api.py

```diff
@@ -0,0 +1,172 @@
+from django.test import TestCase
+from django.urls import reverse
+
+from rest_framework.test import APIClient
+from rest_framework import status
+
+from core.models import Recipe, Ingredient
+
+from recipe.serializers import RecipeSerializer, RecipeDetailSerializer
+
+RECIPES_URL = reverse('recipe:recipe-list')
+
+
+def detail_url(recipe_id):
+    """Return recipe detail url"""
+    return reverse('recipe:recipe-detail', args=[recipe_id])
+
+
+def sample_ingredient(recipe, name='love'):
+    """Create and return a sample ingredient"""
+    return Ingredient.objects.create(recipe=recipe, name=name)
+
+
+def sample_recipe(**params):
+    """Create and return a sample recipe"""
+    defaults = {
+        'name': 'Unicorn cakes',
+        'description': 'Sparkly, swirly and bright',
+    }
+    defaults.update(params)
+    recipe = Recipe.objects.create(**defaults)
+    sample_ingredient(recipe=recipe)
+    return recipe
+
+
+class RecipeApiTests(TestCase):
+    """Test recipe API"""
+
+    def setUp(self):
+        self.client = APIClient()
+
+    def test_retrieve_recipes(self):
+        """Test retrieving a list of recipes"""
+        sample_recipe()
+        sample_recipe()
+
+        response = self.client.get(RECIPES_URL)
+
+        recipes = Recipe.objects.all().order_by('-id')
+        serializer = RecipeSerializer(recipes, many=True)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data, serializer.data)
```

Comment:
Better test against the raw data. The serializer is part of the API and you don't test it like that

Reply:
👍  I'll have a look at improving this test